File: 프로그래머스/2/42586. 기능개발/기능개발.py
# pregresses와 speed의 맨 앞 데이터를 꺼낸다
# 완성 가능일을 계산한다 (99 - progress) // speed + 1
# 다음 데이터를 꺼낸 후 완성 가능일을 계산한다
# 이전 데이터보다 짧다면 complete += 1
# 길다면 answer.append(complete)
def solution(progresses, speeds):
    answer = [0]
    
    pre_complete_day = (99 - progresses[0]) // speeds[0] + 1
    for progress, speed in zip(progresses, speeds):
        complete_day = (99 - progress) // speed + 1
        
        if complete_day <= pre_complete_day:
            answer[-1] += 1
        else:
            answer.append(1)
            pre_complete_day = complete_day

        
        
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <string>
#include <vector>
#include <cmath>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    
    int count = 0;
    int pre_day = ceil((double)(100 - progresses[0]) / speeds[0]);
    for(int i = 0; i < progresses.size(); i++){
        // day가 같거나 작다면 count++
        // day가 크다면 count 추가 및 day 엎어쓰기 count = 1
        int now_day = ceil((double)(100 - progresses[i]) / speeds[i]);
        if (pre_day >= now_day){
            count++;
        } else {
            answer.push_back(count);
            pre_day = now_day;
            count = 1;
        }
    }
    answer.push_back(count);
    
    
    return answer;
}